satpy/tests/reader_tests/test_seviri_l1b_hrit.py

```diff
@@ -450,3 +482,57 @@ def test_calibrate(
 
         res = fh.calibrate(counts, calibration)
         xr.testing.assert_allclose(res, expected)
+
+    @pytest.mark.parametrize(
+        ('channel', 'calibration', 'calib_mode', 'use_ext_coefs'),
+        [
+            # VIS channel, internal coefficients
+            ('VIS006', 'counts', 'NOMINAL', False),
+            ('VIS006', 'radiance', 'NOMINAL', False),
+            ('VIS006', 'radiance', 'GSICS', False),
+            ('VIS006', 'reflectance', 'NOMINAL', False),
+            # VIS channel, external coefficients (mode should have no effect)
+            ('VIS006', 'radiance', 'GSICS', True),
+            ('VIS006', 'reflectance', 'NOMINAL', True),
+            # IR channel, internal coefficients
+            ('IR_108', 'counts', 'NOMINAL', False),
+            ('IR_108', 'radiance', 'NOMINAL', False),
+            ('IR_108', 'radiance', 'GSICS', False),
+            ('IR_108', 'brightness_temperature', 'NOMINAL', False),
+            ('IR_108', 'brightness_temperature', 'GSICS', False),
+            # IR channel, external coefficients (mode should have no effect)
+            ('IR_108', 'radiance', 'NOMINAL', True),
+            ('IR_108', 'brightness_temperature', 'GSICS', True),
+            # HRV channel, internal coefficiens
+            ('HRV', 'counts', 'NOMINAL', False),
+            ('HRV', 'radiance', 'NOMINAL', False),
+            ('HRV', 'radiance', 'GSICS', False),
+            ('HRV', 'reflectance', 'NOMINAL', False),
+            # HRV channel, external coefficients (mode should have no effect)
+            ('HRV', 'radiance', 'GSICS', True),
+            ('HRV', 'reflectance', 'NOMINAL', True),
+        ]
+    )
+    def test_mask_bad_quality(
+            self, file_handler, counts, channel, calibration, calib_mode,
+            use_ext_coefs
+    ):
+        """Test the masking of bad quality scan lines."""
+        external_coefs = self.external_coefs if use_ext_coefs else {}
+        expected = self._get_expected(
+            channel=channel,
+            calibration=calibration,
+            calib_mode=calib_mode,
+            use_ext_coefs=use_ext_coefs
+        )
+
+        fh = file_handler
+        fh.channel_name = channel
+        fh.calib_mode = calib_mode
+        fh.ext_calib_coefs = external_coefs
+
+        res = fh._mask_bad_quality(expected)
```

Comment:
Not really. To adhere to unit test best practice this test is only testing this method explicitly (like the test for the calibration above), which I can not do if I use the `get_dataset` method because there are other things happening apart from the masking. If I remember correctly the method was public at first like the `calibrate` method but I think I made this method private based on a wish from @mraspaud. I agree though that it makes sense that it is private. Anyway in this case the "tight coupling" is not a big problem I suppose.

Reply:
Ok, fine by me :)